clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -180,6 +186,25 @@ public class AsyncKafkaConsumer<K, V> implements ConsumerDelegate<K, V> {
      */
     private class BackgroundEventProcessor implements EventProcessor<BackgroundEvent> {
 
+        private Optional<StreamsRebalanceListener> streamsGroupRebalanceCallbacks = Optional.empty();
+        private final Optional<StreamsRebalanceData> streamsRebalanceData;
+
+        public BackgroundEventProcessor() {
+            this.streamsRebalanceData = Optional.empty();
+        }
+
+        public BackgroundEventProcessor(final Optional<StreamsRebalanceData> streamsRebalanceData) {
+            this.streamsRebalanceData = streamsRebalanceData;
+        }
+
+        private void setStreamsRebalanceListener(final StreamsRebalanceListener streamsRebalanceListener) {
```

Comment:
I think you mentioned this, but could we not just make `streamsRebalanceListener` a meber of `streamsRebalanceData`? If not, could we pass it in the constructor instead of using a setter?